### ISSUE TITLE
Chain Versionable - Occasional ThreadLocal issue on check in

### ISSUE THREAD

**Issue by jharkins2** [tsgrp/hpi#2443]
Occasionally we are seeing the following error occur during the check in of CVN documents

```
java.lang.RuntimeException: ThreadLocal variable contains null
	at com.tsgrp.alfresco.chained.versionable.VersionChainPolicy.createVersionCopy(VersionChainPolicy.java:322)
	at com.tsgrp.alfresco.chained.versionable.VersionChainPolicy.access$100(VersionChainPolicy.java:48)
	at com.tsgrp.alfresco.chained.versionable.VersionChainPolicy$1.doWork(VersionChainPolicy.java:212)
	at com.tsgrp.alfresco.chained.versionable.VersionChainPolicy$1.doWork(VersionChainPolicy.java:181)
	at org.alfresco.repo.security.authentication.AuthenticationUtil.runAs(AuthenticationUtil.java:602)
	at com.tsgrp.alfresco.chained.versionable.VersionChainPolicy.beforeCheckIn(VersionChainPolicy.java:181)
```

This appears to be a timing issue

**Comment by jharkins2** [comment]
Closed at rev #10487

CVN CHECKIN

Fixing ThreadLocal issue which would occasionally occur on check in. We are no longer using a ThreadLocal to keep track of document names

CR: aking, dgrumieaux

GH: https://github.com/tsgrp/HPI/issues/2443

**Comment by katiebrill** [comment]
commit 10512 

fix dependency issues.
